performance_curves/performance_curve.py

```diff
@@ -1,35 +1,134 @@
+import operator
 import numpy as np
 import matplotlib.pyplot as plt
 from typing import Callable, Optional
 
 from performance_curves.utils import synchronize_sort
 
 
-class PerformanceCurve:
+class PerformanceCurveLike:
     """Stores values of the x- and y-axis of the performance curve and allows plotting"""
-    def __init__(self, cutoff, metric_value):
-        self.cutoff = cutoff
+    def __init__(self, metric_value, example_counts):
         self.metric_value = metric_value
+        self.example_counts = example_counts
 
     def plot(self):
         fig, ax = plt.subplots()
-        ax.plot(self.cutoff, self.metric_value)
-        ax.set_xlim([1., 0.])
-        ax.set_ylim([0., 1.10])
-        ax.set_xticks([i for i in reversed(np.arange(0, 1, 0.1))])
-        ax.set_yticks([i for i in np.arange(0, 1.01, 0.1)])
+        ax.plot(self.example_counts, self.metric_value)
+        ax.set_xlabel('Number of Evaluated Cases Ranked in Descending Order')
+        ax.set_ylabel('Metric Value')
+        ax.set_title('Performance Curve')
+        plt.show()
+
+
+class PerformanceCurve(PerformanceCurveLike):
+    def __init__(self, metric_value, example_counts, score_cutoff):
+        super().__init__(metric_value, example_counts)
+        self.score_cutoff = score_cutoff
+
+    def plot_with_cutoff(self):
+        fig, ax = plt.subplots()
+        ax.plot(self.score_cutoff, self.metric_value)
         ax.set_xlabel('Probability Estimates in Descending Order')
         ax.set_ylabel('Metric Value')
         ax.set_title('Performance Curve')
         plt.show()
 
+    def cutoff_at(
+            self,
+            performance_point: float,
+            comparison_operator: str,
+            count_upper_bound: Optional[int] = None,
+            count_lower_bound: Optional[int] = None,
+            cutoff_upper_bound: Optional[float] = None,
+            cutoff_lower_bound: Optional[float] = None,
+            highest: bool = False
+    ) -> float:
```

Comment:
Interesting redesign into a single function. Some questions and things to think about:

- Does changing the comparison operator meaningfully change the result? In my mental model, we are trying to identify score cutoffs where the model's performance crosses the desired performance objective. 

- I can see how using a single function with the `highest` argument might DRY out the code a bit. There might be other ways to accomplish the same thing that maintain function names with clear semantics.